src/main/java/org/javamodularity/moduleplugin/tasks/PackageScanner.java

```diff
@@ -0,0 +1,65 @@
+package org.javamodularity.moduleplugin.tasks;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for retrieving and collecting package names from source files (Java, Kotlin, Groovy, or Scala).
+ */
+public class PackageScanner {
+    private static final Logger LOGGER = Logging.getLogger(PackageScanner.class);
+
+    private static final String WS = "[ \t\r\n\\u000C]";
+    private static final String LETTER = "[a-zA-Z$_]|[^\\u0000-\\u007F\\uD800-\\uDBFF]|[\\uD800-\\uDBFF]|[\\uDC00-\\uDFFF]";
+    private static final String LETTER_OR_DIGIT = LETTER + "|[0-9]";
+    private static final String IDENTIFIER = "((" + LETTER + ")(" + LETTER_OR_DIGIT + ")*)";
+    private static final String QUALIFIED_NAME = IDENTIFIER + "(\\." + IDENTIFIER + ")*";
+
+    private static final String LINE_COMMENT = "//[^\r\n]*";
+    private static final String MULTILINE_COMMENT = "/\\*.*?\\*/";
+    private static final String IGNORE =  "(" + WS + "|" + LINE_COMMENT + "|" + MULTILINE_COMMENT + ")*";
+
+    private static final String PACKAGE_DECLARATION = "(?s)" + IGNORE + "package" + IGNORE + "(?<PACKAGE>" + QUALIFIED_NAME + ").*?";
+
+    private static final Pattern PATTERN = Pattern.compile(PACKAGE_DECLARATION);
```

Comment:
Have you measured the performance of this regexp? How would it perform on a large code base?

Reply:
I didn't measure it yet. It may be an issue.